src/Accelerators/NNPA/Conversion/ONNXToZHigh/RewriteONNXForZHigh.cpp

```diff
@@ -397,6 +473,10 @@ void RewriteONNXForZHighPass::runOnOperation() {
     return true;
   });
 
+  target.addDynamicallyLegalOp<ONNXConvOp>([](ONNXConvOp op) {
+    return !op.pads() || !hasNonZeroInArrayAttr(op.pads().value());
```

Comment:
NNPA supports several cases of non-zero pads. But it looks like this will rewrite ALL conv ops that have non-zero pads no matter these pads are supported by NNPA or not. So, the later pass `convert-onnx-to-zhigh` has no chance to lower supported non-zero padding op directly to ZHigh. Note that `rewrite-onnx-for-zhigh` is called before `convert-onnx-to-zhigh`.

To avoid that, we need to include in here the legalization condition for Conv in ONNXLegalityCheck.cpp: https://github.com/onnx/onnx-mlir/blob/main/src/Accelerators/NNPA/Conversion/ONNXToZHigh/ONNXLegalityCheck.cpp#L778, and maybe more to handle dynamic/static cases.

Reply:
Sure! I am fixing it.

Reply:
Fixed. Thank you!